src/main/helpers/proxy.helpers.ts

```diff
@@ -0,0 +1,75 @@
+import log from 'electron-log';
```

Comment:
Can this logic be in `request.service.ts` instead since `proxy.helper.ts` is just only used by that specific singleton class, no point of exposing proxy stuff outside request/network code.

Reply:
Personally I would prefer making it as standalone as possible.

As of its concept, it is much closer to a system specific helper module like `hardware acceleration toggle`, instead of a sub-module of `request`. Besides, `request` does not require `proxy` to work.

Since it does not only affect network code in `request`, I would prefer moving the entry to `main.ts`, just below `hardware acceleration`. I think it's much better than invading/injecting it into `request`.

Reply:
Hi, I can confirm moving the entry to `main.ts` solves the following two connection failure due to network issue.

It looks like a better solution than making it initialize in `request`.

```log
[2024-12-24 21:28:45.127] [warn]  Download failed over IPv4 for URL: https://raw.githubusercontent.com/Zagrios/beat-saber-scraped-maps/master/song_details_cache_v1.gz RequestError: read ECONNRESET
    at ClientRequest.<anonymous> (D:\GitRepos\bs-manager\release\build\win-unpacked\resources\app.asar\dist\main\main.js:14:1629331)
    at Object.onceWrapper (node:events:634:26)
    at ClientRequest.emit (node:events:531:35)
    at ClientRequest.emit (node:domain:488:12)
    at emitErrorEvent (node:_http_client:108:11)
    at TLSSocket.socketErrorListener (node:_http_client:511:5)
    at TLSSocket.emit (node:events:519:28)
    at TLSSocket.emit (node:domain:488:12)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:218:20)
```

```log
[2024-12-24 22:07:09.251] [warn]  IPv4 request failed, trying next one... URL: https://raw.githubusercontent.com/Zagrios/bs-manager/master/assets/jsons/bs-versions.json RequestError: connect ETIMEDOUT 185.199.110.133:443
    at ClientRequest.<anonymous> (D:\GitRepos\bs-manager\release\build\win-unpacked\resources\app.asar\dist\main\main.js:14:1586333)
    at Object.onceWrapper (node:events:634:26)
    at ClientRequest.emit (node:events:531:35)
    at emitErrorEvent (node:_http_client:108:11)
    at TLSSocket.socketErrorListener (node:_http_client:511:5)
    at TLSSocket.emit (node:events:519:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)
[2024-12-24 22:07:09.260] [warn]  IPv6 request failed, trying next one... URL: https://raw.githubusercontent.com/Zagrios/bs-manager/master/assets/jsons/bs-versions.json RequestError: getaddrinfo ENOENT raw.githubusercontent.com
    at ClientRequest.<anonymous> (D:\GitRepos\bs-manager\release\build\win-unpacked\resources\app.asar\dist\main\main.js:14:1586333)
    at Object.onceWrapper (node:events:634:26)
    at ClientRequest.emit (node:events:531:35)
    at emitErrorEvent (node:_http_client:108:11)
    at TLSSocket.socketErrorListener (node:_http_client:511:5)
    at TLSSocket.emit (node:events:519:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26)
[2024-12-24 22:07:09.260] [error] IPv4 and IPv6 requests failed for URL: https://raw.githubusercontent.com/Zagrios/bs-manager/master/assets/jsons/bs-versions.json
```